src/parser.cpp

```diff
@@ -493,6 +499,9 @@ namespace ada::parser {
     // Define parsed URL
     ada::url url = ada::url();
 
+    // most input strings will be ASCII which enables many optimizations.
+    const bool is_ascii = 128>(std::reduce(user_input.begin()+1, user_input.end(), uint8_t(user_input[0]), std::bit_xor<uint8_t>()));
```

Comment:
Really nice. I was looking for something similar in the path state. Nice to find fast paths for common cases

Reply:
I am not sure it is at the right location. If we can't make use of it elsewhere, then it is maybe better to move it inside the parse_host. I am not sure. However, we could quickly determine if we need percent encoding at all, and that's maybe something that could accelerate a bunch of code.

Baby steps.